src/main/java/com/amazon/aws/iot/greengrass/component/common/TemplateParameter.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.aws.iot.greengrass.component.common;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@JsonDeserialize(builder = TemplateParameter.TemplateParameterBuilder.class)
+public class TemplateParameter {
```

Comment:
A parameter needs a default too, right? Seems to make more sense to me to put the default in this section instead of a separate section.

Reply:
this is a point I'd like to discuss. I don't think there's a case where a template has configuration, since everything is being handled by the transformer jar. But for readability from the end-user's perspective I think it's useful to separate very clearly the concepts of default config for components and default config for template parameters. One is for the component itself and one is for components other than itself.

Reply:
How are they not separate? Putting the defaults in with the schema is much more clear IMO than a separate section.

Reply:
Oh I see what you're saying. Sure, that's reasonable. I'll do that.